pkg/nixpath/nixpath_test.go

```diff
@@ -39,3 +41,27 @@ func TestNixPath(t *testing.T) {
 		assert.Error(t, err)
 	})
 }
+
+func TestNixPathAbsolute(t *testing.T) {
+	t.Run("simple (foo)", func(t *testing.T) {
+		s := nixpath.Absolute("foo")
+		assert.Equal(t, nixpath.StoreDir+"/"+"foo", s)
+	})
+	t.Run("subdir (foo/bar)", func(t *testing.T) {
+		s := nixpath.Absolute("foo/bar")
+		assert.Equal(t, nixpath.StoreDir+"/"+"foo/bar", s)
+	})
+	t.Run("with ../ getting cleaned (foo/bar/.. -> foo)", func(t *testing.T) {
+		s := nixpath.Absolute("foo/bar/..")
+		assert.Equal(t, nixpath.StoreDir+"/"+"foo", s)
+	})
+	// test you can use this to exit nixpath.StoreDir
+	// Note path.Join does a path.Clean already, this is only
+	// written for additional clarity.
+	t.Run("leave storeDir", func(t *testing.T) {
+		s := nixpath.Absolute("..")
+		assert.Equal(t, path.Clean(path.Join(nixpath.StoreDir, "..")), s)
+		assert.False(t, strings.HasPrefix(s, nixpath.StoreDir),
+			"path shouldn't have the full storedir as prefix anymore (/nix)")
```

Comment:
This is written a bit more clunky than it might need to be - however I want to support eventually building `go-nix` with another value of `nixpath.StoreDir` constant, and don't want to break the tests in that case.